internal/sync/syncer.go

```diff
@@ -393,3 +409,51 @@ func updateCommitStatus(
 	}
 	return nil
 }
+
+// GetStateCache is the exported method for getting the cache  in other packages
+func (s *Syncer) GetStateCache() LiveStateCache {
+	return s.stateCache
+}
+
+// GetLiveManagedObjects retrieves live managed objects from the provided cache for the given GitSync configuration.
+// It returns a map of Kubernetes resource keys to unstructured objects and an error if any.
+func GetLiveManagedObjects(cache LiveStateCache, gitSync *v1alpha1.GitSync) (map[kube.ResourceKey]*unstructured.Unstructured, error) {
```

Comment:
nit: Why do we need to add these getter functions and expose it externally?

Reply:
yeah it was not required ,removed it.